compiler/src/output_stream.rs

```diff
@@ -26,13 +34,21 @@ impl From<CodeObjectStream> for CodeObject {
 
 impl OutputStream for CodeObjectStream {
     fn emit(&mut self, instruction: Instruction, location: Location) {
+        let effect = stack_effect(&instruction);
+        self.current_stack_size += effect;
+        if self.current_stack_size > self.max_stack {
+            self.max_stack = self.current_stack_size;
+        }
```

Comment:
You could get rid of the properties on `CodeObjectStream` and just mutate `max_stack_size` on `CodeObject`:

```rust
let max_stack_size = &mut self.code.max_stack_size;
*max_stack_size = std::cmp::max(*max_stack_size, *max_stack_size + effect);
```

Reply:
I understand. The thing is, I would like to gather all instructions in the stream, and in the end, create the immutable codeobject. This makes the code easier to understand in my opinion. That is the reason why I implemented the code like it is.